MongoLog.cc

```diff
@@ -38,10 +49,12 @@ void MongoLog::Flusher() {
   }
 }
 
-std::string MongoLog::FormatTime(struct tm* date) {
-  std::stringstream s;
-  s <<std::put_time(date, "%F %T");
-  return s.str();
+std::string MongoLog::FormatTime(struct tm* date, int ms) {
+  std::string out("YYYY-MM-DD HH:mm:SS.SSS");
+  // this step is technically illegal: "Modifying the character array accessed through the const overload of data has undefined behavior." but I can't c++20 and it worked when tested, so :shrug:
```

Comment:
Just to make sure I understand correctly. We want to make a string "out" and fill it with the date stuff. This is done by accessing "out" through .data and modifying it. And this is undefined behavior

Reply:
But according to here: https://stackoverflow.com/questions/63782107/is-it-safe-to-use-stdstringc-str-to-modify-the-underlying-stdstring this is actually the correct way?

Reply:
Ok, so before c++17 the `.data()` method returned `const char*` and modifying it was undefined, but after c++17 it returns `char*` so as long as we don't write past the end (which we shouldn't) we should be fine.